### PR TITLE
Add 'dingbats' and 'misc symbols' ranges

### PR DESCRIPTION
**Pull request by bk138**
This commit adds the https://en.wikipedia.org/wiki/Miscellaneous_Symbols and https://en.wikipedia.org/wiki/Dingbat#Unicode ranges to the scanned ranges, enabling entry of symbols like ✅ and ⛔.

### CODE REVIEW

**Comment by Sbozzolo** [comment]
Thanks. I feel conflicted about this change: I clearly agree in adding those codepoints that enable emojis like :white_check_mark: and :no_entry:, but I feel that this change also brings in a lot of less useful symbols which we would not traditionally call emojis. 

My inclination would be to add to the defaults only those symbols in the table https://en.wikipedia.org/wiki/Dingbat#Unicode that are commonly recognized as emojis and found in keyboards in smartphones. If users want the full range of dingbats, they can customize `ivy-emoji-codepoint-ranges` to their need. 

What do you think? Do you often use non-emoji symbols?

**Comment by bk138** [comment]
Thanks for the quick reply, I see your point about focusing on emoji. Maybe a better approach would be to see which ranges other tools show to users. I personally use https://github.com/maoschanz/emoji-selector-for-gnome and wanted something similar directly in Emacs. From dingbats, for instance, it shows ✅ but not ✠. Maybe it's worth looking there how they do it and adapt this for ivy-emoji? I can have a look if that would be an OK approach for you.

**Comment by Sbozzolo** [comment]
[This](https://git.savannah.gnu.org/cgit/emacs.git/commit/?id=6b706b1a7d40faa10989ea3419e5cda0067ef82a) is how Emacs is going to support emojis in 29.1. Essentially, there's a a [text file](https://git.savannah.gnu.org/cgit/emacs.git/diff/admin/unidata/emoji-test.txt?id=6b706b1a7d40faa10989ea3419e5cda0067ef82a) that contains the list of characters that are emoji. This is also the approach taken by `telega.el`. This is an option I took in consideration when developing `ivy-emoji`. The reason I ended up working with ranges is because it keeps the source code clean and simple. In doing that, I also made sure to keep `ivy-emoji-list` customizable, so that one could use lists provided by other packages or Emacs itself.

In other words, I feel the current approach is simple but effective, so I don't have see any urgent need for improvements. That said, I certainly want to provide the best out-of-the-box experience possible, so I am open to suggestions/improvements. (For example, a low-hanging fruit might be to check if `emacs-version >= 29` and use the official emoji list instead of looking at ranges.)

**Comment by bk138** [comment]
Alright, so Emacs 29 will have a specific list of emoji (very much like telega or company-emoji have it now). My suggestion is that ivy-emoji should just that as a default if it's available.

I understand that you want to keep ivy-emoji's code simple, OTOH my feedback as a user is that the range scanning as it is now does not yield the best UX: some symbols like the mentioned ✅ and ⛔ are not included whereas some others like `mahjong-tile...` are ;-)

To have  a decent user experience, allow me to suggest
* using company-emoji's symbol list for Emacs <= 28 (which I found very close to what one would expect after using a mobile messenger)
* using Emacs's built-in list for Emacs >= 29.

I understand this probably means some work, I can't provide a PR though as my Lisp skills are severely lacking, so am totally OK if you go for just one of the options or none.

Thanks for this useful package, I'll close this specific PR then.

**Comment by Sbozzolo** [comment]
I agree that the range selection is very imperfect and that the countless variants of the `mahjong-tiles` or `playing-card` are nearly useless. 

So, this is my suggestion:
- if Emacs>=29.1, set `ivy-emoji-list` to the built-in list
- If Emacs<=29.1, check we can add a list of packages with known curated emoji lists (`company-emoji`, `telega`, `emojify`, ...) is available and use that list. If none of the packages is available, we construct our list parsing ranges. I'd say that it is better to be overly inclusive instead of excluding some potentially useful symbols (at the end, `ivy-emoji` is very fast, so having a long list should not be a big problem...)

If you want to work on your Elisp skills, this might be a nice problem. This is the sketch of the solution:
- The default list of emojis is defined by evaluating `(ivy-emoji--create-list)`, so `ivy-emoji--create-list` is the function that needs to be adjusted.
- (The double hyphen indicates that this is a private method, by convention)
- `ivy-emoji--create-list` returns a list with emojis and their names, with the following format: `🌵 :cactus:`
- You can check if a package is available with `fboundp`, specifically, you can check directly if the emoji list defined by the package is available and not empty. 
- Different packages store emojis in different ways, but `ivy-emoji-create-list` has to return a list as described above. So, for example, `telega` has a variable `telega-emoij-alist` which is a list of associative lists with individual elements of the form `(":cactus:" . "🌵")`. We would need to massage this to be in the format mentioned before. 

Happy to provide more directions if you want to try this out!

**Comment by bk138** [comment]
Thanks for the thorough explanation. I'll put it on my "hobby projects" agenda, but am currently quite busy with €€€-work and family, so don't expect anything too soon ;-)
